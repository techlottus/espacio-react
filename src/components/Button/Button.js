

import React, { useEffect } from 'react'


export const Button = React.memo(({ data, onClick }) => {



  const buttonRef = React.createRef();
  
  useEffect(() => {
    buttonRef.current.data = {
      id: data.id || null,
      type: data.type  || 'primary',
      title: data.title || '',
      size: data.size || 'small',
      icon: data.icon || '',
      lyIcon: data.lyIcon,
      disabled: data.disabled,
      isExpand: data.isExpand,
      test: data.test || ''
    }
  }, [data]) // eslint-disable-line react-hooks/exhaustive-deps

  useEffect(() => {
    buttonRef.current.addEventListener('onClick', onClick);
  }, []) // eslint-disable-line react-hooks/exhaustive-deps

  return (
    <>
      <lottus-button ref={buttonRef}></lottus-button>
    </>
  )
})


