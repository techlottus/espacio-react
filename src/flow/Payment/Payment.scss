@import '../../App.scss';

.payment-container {
  position: relative;
  width: 100%;
  font-family: 'Nunito', sans-serif;
  display: flex;
  flex-direction: column;
  @extend .headerapp-top;
  color: map-get($palette-neutral, n80);
  margin-bottom: pixeltorem(250px);


  @media only screen and (min-width: map-get($grid-breakpoints, m)) {
    flex-direction: row;
    margin-bottom: 0;
  }

  .paymentstages {
    width: calc(100% - #{pixeltorem(32px)});
    padding: pixeltorem(36px) pixeltorem(16px);

    @media only screen and (min-width: map-get($grid-breakpoints, m)) {
      width: calc(50% - #{pixeltorem(72px)});
      padding: pixeltorem(36px) pixeltorem(24px) pixeltorem(36px) pixeltorem(48px);
    }
  
    .paymenttitle {
      margin-top: pixeltorem(36px);
      margin-bottom: pixeltorem(36px);
      font-size: pixeltorem(24px);
      font-family: 'Nunito Bold', sans-serif;
  
      .paymenticon {
        margin-left: pixeltorem(8px);
      }
    }
  
    .paymentsteps {
      width: 100%;
      height: pixeltorem(100px);
      margin-bottom: pixeltorem(40px);
    }
  
    .paymentcontent {
      width: 100%;
      margin-bottom: pixeltorem(180px);

      > .paymentcontentchild {
        width: 100%;
        height: 100%;
      }

      @media only screen and (min-width: map-get($grid-breakpoints, m)) {
        margin-bottom: 0;
      }
    }
  }

  .paymentresume {
    right: 0;
    position: fixed;
    display: none;
    width: 50%;
    @extend .fixedcontainer-height;

    @media only screen and (min-width: map-get($grid-breakpoints, m)) {
      display: block;
    }

  }
  
}

