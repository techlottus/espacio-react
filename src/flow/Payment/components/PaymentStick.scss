@import '../../../App.scss';

.paymentstick-container {
  font-family: "Nunito", sans-serif;
  position: fixed;
  z-index: 10;
  left: 0;
  right: 0;
  bottom: 0;
  width: calc(100% - #{pixeltorem(36px)});
  height: pixeltorem(100px);
  padding: 0 pixeltorem(18px);
  display: flex;
  flex-direction: column;
  background-color: map-get($palette-neutral, n00);
  border-radius: pixeltorem(30px) pixeltorem(30px) 0 0;
  box-shadow: 0 pixeltorem(-5px) pixeltorem(30px) rgba(0, 0, 0, 0.15);

  @media only screen and (min-width: map-get($grid-breakpoints, m)) {
    display: none;
  }

  .paymentsticksecure {
    width: 100%;
    margin-top: pixeltorem(24px);
    margin-bottom: pixeltorem(36px);
  }

  > .icon {
    cursor: pointer;
    margin-top: pixeltorem(12px);
    font-size: pixeltorem(30px);
    text-align: center;
    color: palette-colors($brand, p00);
  }

  > .paymentstickempty {
    width: 100%;
    margin-top: pixeltorem(8px);

    .paymentsticktitle {
      font-family: "Nunito Bold", sans-serif;
      font-size: pixeltorem(18px);
      line-height: pixeltorem(24px);
    }

    .paymentsticktsubtitle {
      font-size: pixeltorem(12px);
      line-height: pixeltorem(16px);
    }
  }

  .paymentstickcard {
    margin-top: pixeltorem(24px);
  }
}

.modaltextsticky {
  font-size: pixeltorem(14px);
  margin-bottom: pixeltorem(16px);
}

.modaltextstickinvoice {
  font-size: pixeltorem(12px);
}

.modalinvoicesticky {
  font-size: pixeltorem(12px);
  margin-bottom: pixeltorem(16px);
}

.totalmodalsticky {
  margin-bottom: pixeltorem(16px);
}

.modalstickybtn {
  button {
    width: 100%;
    justify-content: center;
    text-align: center;
  }

  .backstickybtn {
    margin-bottom: pixeltorem(15px);
  }
}
