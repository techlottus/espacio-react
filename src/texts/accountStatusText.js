export const accountStatusText = {
  statusTitle: "Estado de cuenta y consultas de facturas",
  resumeTitle: "Resumen de adeudo",
  total: "Total",
  errorMessage:
    "Tuvimos dificultades para cargar tu información en este momento",
  emptyMessage: {
    congrats: "Felicidades, estás al día con tus pagos",
    continue: "Continúa transformando tu futuro",
  },
  images: { 
    user: "images/user.png",
    empty: "yeah.png",
  },
  btnModal: {
    delete: "Eliminar",
    cancel: "Cancelar",
  },
  btnDownload: {
    download: "Descargar Estado de cuenta",
  },
  btnInvoice: {
    checkInvoice: "Consulta tus facturas",
  },
  tableAccountStatus: [
    { title: "Clave" },
    { title: "Ciclo" },
    { title: "Periodo" },
    { title: "Adeudo" },
  ],

  periodDetailTitle: "Detalles por periodo",
  periodDetailDescription:
    "Selecciona el programa y el periodo para ver los detalles de los movimientos; se clasifican de la siguiente manera:",
  periodDetailBullets: {
    debt: "Con adeudo",
    payed: "Pagados",
    toPay: "Por pagar",
  },
  periodDetailSelects: {
    program: "Programa",
    period: "Periodo",
  },
  accountDetailPeriod: {
    title: "Periodo",
    icon: "calendar_today",
    debt: "Saldo del periodo",
  },
  accountDetailMovements: {
    title: "Movimientos",
  },
  tableInvoiceQuery: [
    { title: "Fecha de emisión" },
    { title: "Periodo" },
    { title: "Concepto" },
    { title: "Monto" },
    { title: "RFC" },
    { title: "Preview" },
    { title: "PDF" },
    { title: "XML" },
  ],
  tableRowInvoiceQuery: [
    { title: "07 Abril 2021" },
    { title: "Periodo 5" },
    { title: "Colegiatura" },
    { title: "$2405.50" },
    { title: "LRKO485767HF2" },
  ],
  accountStatusInvoiceQuery: {
    title: "Consulta de facturas",
    description:
      "Filtra por rango de fechas, selecciona el perido que deseas consultar y presiona descargar.",
  },
  btnInvoiceQuery: {
    download: "Descargar todo",
    clear: "Limpiar",
  },
  linkInvoiceQuery: "Ver más"
};
