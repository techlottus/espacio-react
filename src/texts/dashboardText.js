export const dashboardText = {
  menu: {
    aula: {
      emoji: "mortar-board",
      text: "Aula",
    },
    book: {
      emoji: "books",
      text: "Biblioteca",
    },
    helpCenter: {
      emoji: "thought-balloon",
      text: "Centro de Ayuda",
    },
    procedures: {
      emoji: "file-folder",
      text: "Trámites",
    },
    accountStatus: {
      emoji: "chart-with-upwards-trend",
      text: "Estado de Cuenta",
    },
    payment:  {
      emoji: "credit-card",
      text: "Pagos",
    },
    invoice: {
      emoji: "page-facing-up",
      text: "Facturas"
    },
    benefits: {
      emoji: "star",
      text: "Portal de Beneficios"
    },
    sede: {
      emoji: "notebook",
      text: "SeDe"
    },
    psychopedagogical: {
      emoji: "blue-heart",
      text: "Centro de Atención Psicopedagógica"
    },
    evaluateServices: {
      emoji: "ec ec-white-check-mark",
      text: "Evalúa los servicios"
    },
    registrationSubjects: {
      emoji: "ec ec-closed-book",
      text: "Inscribe tus materias"
    },
    logout: {
      emoji: "back",
      text: "Cerrar sesión",
    },
  },
  titleWelcome: "Hola",
  images: {
    cardHeader: "images/header.png",
    empty: "empty.png",
  },
  logo: {
    urlLogo: "logo.svg"
  },
  logos: "logo.svg",
  sectionDashboard: {
    titleNews: "Noticias de tu universidad",
    titleCards: "¿Qué quieres hacer hoy?",
  },
  playIcon: "play_circle_outline",
  cards: {
    one: {
      title: "Aula",
      icon: "school",
    },
    two: {
      title: "Pagos",
      icon: "credit_card",
    },
    three: {
      title: "Biblioteca",
      icon: "import_contacts",
    },
    four: {
      title: "Trámites",
      icon: "post_add",
    },
    five: {
      title: "Inscribe tus materias",
      icon: "add_circle_outline",
    },
  },
  footerText: {
    text: "©2022 Derechos reservados Lottus Education",
    privacy: "Aviso de privacidad y términos y condiciones",
  },
};