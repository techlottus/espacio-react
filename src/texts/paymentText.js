export const paymentText = {
  "stepper": {
    "titleOne": "Concepto",
    "titleTwo": "Datos fiscales",
    "titleThree": "Método de pago"
  },
  "breadcrumb": {
    "breadOne": "Home",
    "breadTwo": "Pagos",
    "icon": "home"
  },
  "invoiceType": {
    "one": "Persona física",
    "two": "Persona moral"
  },
  "addInvoice": "Agregar razón social",
  "forms": {
    "businessName": {
      "title": "*Nombre de la razón social",
      "error": "Nombre de la razón social requerido"
    },
    "rfc": {
      "title": "*RFC",
      "error": "Ingresa un RFC válido"
    },
    "zipCode": {
      "title": "*Código postal",
      "error": "El código postal es requerido"
    },
    "state": {
      "title": "*Estado o provincia"
    },
    "city": {
      "title": "*Ciudad",
      "error": "La ciudad es requerida"
    },
    "neighborhood": {
      "title": "*Colonia",
      "error": "La colonia es requerida"
    },
    "street": {
      "title": "*Calle y número",
      "error": "La calle y el número es requerido"
    },
    "curp": {
      "title": "*CURP del alumno",
      "error": "Ingresa un CURP válido"
    },
    "email": {
      "title": "Correo alterno"
    },
    "cfdiUsage": {
      "title": "*Uso de CFDI"
    },
    "regime": {
      "title": "*Régimen Fiscal"
    },
    "fullName": {
      "title": "*Nombre completo",
      "error": "Nombre requerido"
    }
  },
  "cfdiOptions": {
    "optionOne": "Pagos por servicios educativos",
    "optionsTwo": "Gastos general",
    "optionsThree": "Por definir"
  },
  "listTaxTitle": "La factura se emitará bajo la siguiente información",
  "formTaxTitle": "Completa los datos fiscales",
  "paymentTitle": "Pagos",
  "conceptTitle": "Para comenzar seleciona el concepto que deseas pagar",
  "methodsTitle": "Selecciona el método de pago seguro que más se adapte a tus necesidades",
  "invoiceText": "Deseo facturar",
  paymentMethods: [
    {
      title: "Pago con tarjeta",
      label: " - Pago en línea",
      icon: "credit_card",
      description:
        "Paga en línea con cualquier tarjeta de crédito o débito",
    },
    {
      title: "Ficha de depósito",
      label: " - Pago en efectivo",
      icon: "description",
      description:
        "Descarga aquí tu ficha de referencia, con la cual podrás pagar en Oxxo, Seven Eleven, Farmacias del Ahooro, Telecomm, o en las ventanillas de Banorte, Santander y BBVA.",
    },
  ],
  resume: {
    resumeText: "Resumen de compra",
    title: "No tienes conceptos cargados",
    description: "Selecciona un concepto para ver el resumen",
    total: "Total",
    amount: "Recargo",
    concept: "Concepto",
    payment: "Pagar",
    next: "Continuar",
    selectNotice: "Selecciona los conceptos que deseas pagar",
    imageSelectNotice: "images/utc/icon-card.svg",
    toggleNotice: "No se pueden emitir facturas después del pago",
    invoiceText: "Deseo facturar",
    feedbackNotice: "Completa tu información",
  },
  "stick": {
    "notConcepts": "No tienes conceptos cargados",
    "resumePayment": "Resumen de compra",
    "initResume": "Comienza con tu pago fácil y seguro",
    "addArt": "Agrega los artículos que desea pagar"
  },
  "cardResume": {
    "label": "WA-202180",
    "title": "Licenciatura en Diseño gráfico",
    "subtitle": "Nuevo Ciclo 21-2",
    "price": 2405.5,
    "totalTitle": "Total",
    "totalUnit": 2405
  },
  "btnModal": {
    "next": "Continuar",
    "back": "Regresar",
    "delete": "Eliminar",
    "cancel": "Cancelar",
    "accept": "Aceptar",
    "save": "Guardar",
    "exit": "Salir"
  },
  "btns": {
    "next": "Continuar",
    "back": "Atrás",
    "save": "Guardar"
  },
  "cardInvoice": {
    "address": "Dirección:",
    "email": "Correo electrónico:",
    "businessName": "Razón social:",
    "rfc": "RFC:",
    "name": "Nombre:",
    "curp": "CURP:",
    "cfdi": "Uso de CFDI:",
    "regime": "Régimen Fiscal"
  },
  "actions": {
    "download": "Descargar",
    "payment": "Pagar"
  },
  "concepts": {
    "description": "Vencimiento:",
    "delete": "Eliminar",
    "prefix": "WA"
  },
  "modals": {
    "modalPaymentMethods": {
      "title": "Verifica que tus datos sean correctos",
      "data": {
        "title": "La factura se emitirá bajo los siguientes datos fiscales",
        "rfc": "RFC:"
      },
      "paymentText": "Ingresa el monto que quieres pagar"
    },
    "modalIcon": "help",
    "modalToggleConcept": {
      "title": "No se pueden emitir facturas después de realizar el pago. ¿Deseas realizar tu pago sin solicitar factura?"
    },
    "modalConceptDelete": {
      "title": "¿Seguro que quieres eliminar este concepto?"
    },
    "modalInvoiceDelete": {
      "title": "¿Estás seguro que deseas eliminar estos datos fiscales?"
    },
    "modalInvoice": {
      "title": "¿Son correctos tus datos fiscales?"
    },
    "modalCardDelete": {
      "title": "¿Seguro que quieres eliminar esta tarjeta?"
    }
  },
  "errors": {
    "getInvoice": "Estamos teniendo problemas para procesar datos fiscales, inténtalo más tarde",
    "maxInvoices": "Solo puedes agregar 3 RFC",
    "deleteCard": "No pudimos eliminar la información de la tarjeta, por favor intenta más tarde",
    "postCreditCard": "No se han podido guardar tus datos en este momento, inténtalo más tarde",
    "getCreditCards":  "Tuvimos dificultades para cargar la información, inténtalo más tarde",
    "emptyCreditCards": "No tienes tarjetas guardadas"
  },
  "images": {
    "emptyError": "empty-error.png",
    "emptyFirst": "empty-first.png",
    "visa": "images/pay-visa.svg",
    "american": "images/pay-american.svg",
    "diners": "images/pay-diners.svg",
    "jcb": "images/pay-jcb.svg",
    "mastercard": "images/pay-mastercard.svg",
    "discover": "images/pay-discover.svg",
    "securePayment": "images/security-pay.svg",
    "unknown": "images/pay-unknown.svg"
  },
  "stagesTitles": {
    "one": "Selecciona un concepto",
    "two": "Completa tus datos fiscales",
    "three": "Selecciona un método de pago",
    "four": "Selecciona una tarjeta"
  },
  "secure": {
    "title": "Pago 100% seguro",
    "description": "Tus tarjetas se guardan de forma segura para que puedas reutilizar el método de pago."
  },
  "success":{
    "postCreditCard":"Los datos de la tarjeta fueron guardados correctamente",
    "deleteCard": "Los datos de la tarjeta fueron eliminados correctamente"
  },
  "ulaFeedbackText": "Para facturar entra al menú superior de la derecha en la opción de facturas, después de hacer tu pago"

}