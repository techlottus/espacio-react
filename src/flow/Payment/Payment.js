import React, { useState, useEffect } from "react";
import "./Payment.scss";
import { Stepper } from "../../components/Stepper/Stepper";
import {
  stepper,
  breadcrumb,
  stagesSteps,
  stepsCardResume,
  conceptToogle,
} from "../../constants/Payment.constant";
import { Breadcrumb } from "../../components/Breadcrumb/Breadcrumb";
import PaymentRoutes from "../../routes/PaymentRoutes";
import HeaderApp from "../../shared/HeaderApp/HeaderApp";
import PaymentStick from "./components/PaymentStick";
import { PaymentContext } from "./PaymentContext";
import { typesRequestErrors } from "../../constants/error.constant";
import ScreenError from "../../shared/ScreenError/ScreenError";
import { useDispatch, useSelector } from "react-redux";
import {
  resetPaymentStore,
  setConcept,
  setErrorPayments,
  setPathExitPayment,
} from "../../actions/paymentAction";
import { useHistory, useLocation } from "react-router";
import { ModalPaymentMethods } from "../../components/Modal/ModalPaymentMethods/ModalPaymentMethods";
import { ModalExit } from "../../components/Modal/ModalExit/ModalExit";
import { unblockPayment } from "../../helpers/unblock";
import { getTitleOfPayment } from "../../helpers/getTitlePayment";
import PaymentCardResumeDesk from "./components/PaymentCardResumeDesk";
import env from "../../enviroment/environment";
import { typesMark } from "../../types/typesMark";

const Payment = () => {
  const { texts } = useSelector((state) => state.texts);
  const history = useHistory();
  const location = useLocation();
  const [steps, setSteps] = useState({
    ...stepper,
    titles: [
      {
        text: texts?.payment.stepper.titleOne,
        disabled: false,
      },
      {
        text: texts?.payment.stepper.titleTwo,
        disabled: false,
      },
      {
        text: texts?.payment.stepper.titleThree,
        disabled: false,
      },
    ],
  });

  const [isModalToggle, setIsModalToggle] = useState(false);

  const [toogle, setToogle] = useState({
    ...conceptToogle,
    status: env.mark === typesMark.utc,
  });

  const [isModalExit, setIsModalExit] = useState({
    isShow: false,
    unblock: null,
  });

  const [isModalPayment, setIsModalPayment] = useState(false);
  const [stateCardResume, setStateCardResume] = useState(stepsCardResume.first);

  const dispatch = useDispatch();

  const handleAftherToPayment = (isInvoice) => {
    setSteps({
      ...stepper,
      initCount: 2,
      titles: [
        {
          text: texts?.payment.stepper.titleTwo,
          disabled: !isInvoice,
        },
        {
          text: texts?.payment.stepper.titleThree,
          disabled: false,
        },
        {
          text: "Pago con tarjeta",
          disabled: false,
        },
      ],
      focus: 2,
    });
  };

  const handleBeforeToPayment = (isInvoice) => {
    setSteps({
      ...stepper,
      initCount: 1,
      titles: [
        {
          text: texts?.payment.stepper.titleOne,
          disabled: false,
        },
        {
          text: texts?.payment.stepper.titleTwo,
          disabled: !isInvoice,
        },
        {
          text: texts?.payment.stepper.titleThree,
          disabled: false,
        },
      ],
      focus: 2,
    });
  };

  const handleSteps = (focus, isInvoice) => {
    if (isInvoice) {
      const titleCopy = steps.titles.map((title, index) => {
        return {
          ...title,
          disabled: index === stagesSteps.taxData,
        };
      });
      setSteps({
        ...steps,
        focus,
        titles: titleCopy,
      });
    } else {
      const titleCopy = steps.titles.map((title, index) => {
        return {
          ...title,
          disabled: false,
        };
      });
      setSteps({
        ...steps,
        focus,
        titles: titleCopy,
      });
    }
  };

  const paymentStore = useSelector((state) => state.payment);

  const [isError, setIsError] = useState({
    isError: false,
    msg: "",
  });

  useEffect(() => {
    if (
      paymentStore.errors &&
      paymentStore.errors[typesRequestErrors.getConcepts]?.isError
    ) {
      setIsError({
        isError: true,
        msg: paymentStore.errors[typesRequestErrors.getConcepts]?.msg,
      });
    } else if (
      paymentStore.errors &&
      paymentStore.errors[typesRequestErrors.emptyConcepts]?.isError
    ) {
      setIsError({
        isError: true,
        msg: paymentStore.errors[typesRequestErrors.emptyConcepts]?.msg,
      });
    } else {
      setIsError({
        isError: false,
        msg: null,
      });
    }
  }, [paymentStore.errors]);

  useEffect(() => {
    let unblock = history.block((location) => {
      if (unblockPayment(location.pathname)) {
        dispatch(setPathExitPayment(location.pathname));
        setIsModalExit({
          isShow: true,
          unblock,
        });
        return false;
      }

      return true;
    });

    return () => {
      dispatch(setErrorPayments(typesRequestErrors.getConcepts, false, null));
      dispatch(setErrorPayments(typesRequestErrors.emptyConcepts, false, null));
      dispatch(resetPaymentStore());
      unblock();
    };
  }, []); // eslint-disable-line react-hooks/exhaustive-deps

  const renderSuccess = () => {
    return (
      <div className="paymentcontentchild">
        <div className="paymentsteps">
          <Stepper data={steps} />
        </div>
        <div className="paymentcontent">
          <PaymentRoutes />
        </div>
      </div>
    );
  };

  const renderError = (msg) => {
    return (
      <div style={{ marginTop: "-50px" }}>
        <ScreenError msg={msg} />
      </div>
    );
  };

  const handleStateCardResume = (state) => {
    setStateCardResume(state);
  };

  const handleModalToggle = () => {
    setIsModalToggle((state) => !state);
  };

  const handleToggle = (toogle) => {
    setToogle({ ...toogle });
  };

  const onItem = (e) => {
    if (e.detail === "/") {
      history.push(e.detail);
    }
  };

  const onBack = (e) => {
    history.push("/");
  };

  return (
    <>
      <HeaderApp />
      <PaymentContext.Provider
        value={{
          isModalToggle,
          toogle,
          stateCardResume,
          handleSteps,
          handleBeforeToPayment,
          handleAftherToPayment,
          handleStateCardResume,
          setIsModalPayment,
          handleToggle,
          handleModalToggle,
        }}
      >
        <div className="payment-container">
          <div className="paymentstages">
            <div className="back">
              <Breadcrumb
                data={{
                  ...breadcrumb,
                  textItems: [
                    {
                      ...breadcrumb.textItems[0],
                      text: texts?.payment.breadcrumb.breadOne,
                    },
                    {
                      ...breadcrumb.textItems[1],
                      text: texts?.payment.breadcrumb.breadTwo,
                    },
                  ],
                  icon: texts?.payment.breadcrumb.icon,
                }}
                onItem={onItem}
                onBack={onBack}
              />
            </div>
            <p className="paymenttitle">
              {getTitleOfPayment(
                location.pathname,
                texts?.payment?.stagesTitles
              )}
            </p>
            <div className="paymentcontent">
              {isError.isError ? renderError(isError.msg) : renderSuccess()}
            </div>
            <div className="paymentresume">
              {/* <PaymentResumeDesktop /> */}
              <PaymentCardResumeDesk />
            </div>
          </div>
          <PaymentStick />
        </div>
        <ModalPaymentMethods
          isShow={isModalPayment}
          state={stateCardResume}
          onClose={() => {
            setIsModalPayment((state) => !state);
          }}
          outside={() => setIsModalPayment(false)}
        />
        <ModalExit
          isShow={isModalExit.isShow}
          onClose={() => {
            setIsModalExit((state) => {
              return {
                ...state,
                isShow: false,
              };
            });
          }}
          onNext={() => {
            isModalExit.unblock();
            setIsModalExit({
              isShow: false,
              unblock: null,
            });
            setPathExitPayment(null);
            setConcept(null);
            history.push(paymentStore.pathExit);
          }}
          flow={"payment"}
        />
      </PaymentContext.Provider>
    </>
  );
};

export default Payment;
