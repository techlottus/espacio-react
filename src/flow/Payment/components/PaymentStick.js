import React, {  useContext, useState } from "react";
import { useSelector } from "react-redux";
import "./PaymentStick.scss";
import PaymentCardResumeDesk from "./PaymentCardResumeDesk";
import { PaymentContext } from "../PaymentContext";
import { stepsCardResume } from "../../../constants/Payment.constant";

const PaymentStick = () => {
  const [isShow, setIsShow] = useState(false);
  const paymentStore = useSelector((state) => state.payment);
  const {stateCardResume} = useContext(PaymentContext)

  const calcHeight = () => {
    return  !isShow
    ? paymentStore.selectConcept !== null
      ? stateCardResume === stepsCardResume.second ? '300px': (paymentStore?.searchAmount?.amountAdd ? '300px': '250px')
      : "120px"
    : "520px";
  }

  return (
    <>
      <div
        className="paymentstick-container"
        style={{
          height:calcHeight()
        }}
      >
        <span
          className="material-icons icon"
          onClick={() => {
            setIsShow((state) => !state);
          }}
        >
          {isShow ? "keyboard_arrow_down" : "expand_less"}
        </span>
        <PaymentCardResumeDesk
          isMobile={true}
          isShow={isShow}
        />
      </div>
    </>
  );
};

export default PaymentStick;
