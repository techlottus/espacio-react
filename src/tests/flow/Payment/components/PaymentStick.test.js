import React from "react";
import { mount } from "enzyme";
import { Provider } from "react-redux";
import "@testing-library/jest-dom";
import configureStore from "redux-mock-store";
import thunk from "redux-thunk";
import PaymentStick from "../../../../flow/Payment/components/PaymentStick";
import { PaymentContext } from "../../../../flow/Payment/PaymentContext";
import { Button } from "../../../../components/Button/Button";
import { typesPaymentMethods } from "../../../../constants/Payment.constant";
import { act } from "react-dom/test-utils";
import { paymentText } from "../../../../texts/paymentText";

const middlewares = [thunk];
const mockStore = configureStore(middlewares);
window.dataLayer = {
  push: jest.fn()
}

let initState = {
  payment: {
    selectConcept: {
      description: "test",
      title: "test",
      subtitle: "test",
      price: 20,
    },
    selectMethod: null,
  },
  texts: {
    payment: {
      ...paymentText
    }
  }
};

let store = mockStore(initState);

const setIsModalPayment = jest.fn().mockImplementation(() => {});

const handleMount = (initState) => {
  store = mockStore(initState);
  return mount(
    <PaymentContext.Provider value={{ setIsModalPayment }}>
      <Provider store={store}>
        <PaymentStick />
      </Provider>
    </PaymentContext.Provider>
  );
};

describe("Test <PaymentStick/>", () => {
  beforeEach(() => {
  });

  test("should init component", () => {
    const wrapper = handleMount(initState);
    expect(wrapper).toMatchSnapshot();
  });

  test("should download button", () => {
    initState = {
      ...initState,
      payment: {
        ...initState.payment,
        selectConcept: {
          description: null,
          title: null,
          subtitle: null,
          price: 10, 
        },
        selectMethod: typesPaymentMethods.deposit,
      },
    };
    const wrapper = handleMount(initState);
    expect(wrapper.find(Button).last().props().data.title).toBe("Descargar");
  });

  test("should pay button", () => {
    initState = {
      ...initState,
      payment: {
        ...initState.payment,
        selectConcept: {
          description: null,
          title: null,
          subtitle: null,
          price: 10,
        },
        selectMethod: typesPaymentMethods.mercadoPago,
      },
    };
    const wrapper = handleMount(initState);
    expect(wrapper.find(Button).last().props().data.title).toBe("Pagar");
  });

  test("should empty concept", () => {
    initState = {
      ...initState,
      payment: {
        ...initState.payment,
        selectConcept: null,
      },
    };
    const wrapper = handleMount(initState);
    expect(wrapper.find(".paymentsticktitle").text()).toBe("No tienes conceptos cargados");
  });

  test("should resume concept", () => {
    initState = {
      ...initState,
      payment: {
        ...initState.payment,
        selectConcept: null,
      },
    };
    const wrapper = handleMount(initState);
    act(() => {
      wrapper.find("span.material-icons").props().onClick()
    })
    wrapper.update()
    expect(wrapper.find(".paymentsticktitle").text()).toBe("Resumen de compra");
  });

  test("should select concept", () => {
    initState = {
      ...initState,
      payment: {
        ...initState.payment,
        selectConcept: {
          description: "test",
          title: "test",
          subtitle: "test",
          price: 20,
        },
      },
    };
    const wrapper = handleMount(initState);
    act(() => {
      wrapper.find("span.material-icons").props().onClick()
    })
    wrapper.update()
    expect(wrapper.find(".paymentsticktitle").text()).toBe("Resumen de compra");
  });

  test("should click button", () => {
    const wrapper = handleMount(initState);
    act(() => {
      wrapper.find(Button).first().props().onClick();
    });
    wrapper.update();
    expect(setIsModalPayment).toHaveBeenCalled();
  });
});
