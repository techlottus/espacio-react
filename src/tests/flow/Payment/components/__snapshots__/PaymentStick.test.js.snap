// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <PaymentStick/> should init component 1`] = `
<div
  className="paymentstick-container"
  style={
    Object {
      "height": "180px",
    }
  }
>
  <span
    className="material-icons icon"
    onClick={[Function]}
  >
    expand_less
  </span>
  <div
    className="paymentstickempty"
  >
    <div
      className="paymentsticktitle"
    />
  </div>
  <div
    className="paymenttotalprice"
  >
    <div
      className="paymenttotalpricetitle"
    />
    <div
      className="paymenttotalpriceunit"
    />
  </div>
  <div
    className="paymentstickbtn"
    style={
      Object {
        "marginTop": "12px",
      }
    }
  >
    <lottus-button />
  </div>
</div>
`;
